packages/@tinacms/toolkit/src/packages/fields/plugins/ImageFieldPlugin.tsx

```diff
@@ -68,62 +68,69 @@ export function usePreviewSrc(
   return [src, loading]
 }
 
-export const ImageField = wrapFieldsWithMeta<InputProps, ImageProps>(
-  (props) => {
-    const cms = useCMS()
-    const { form, field } = props
-    const { name, value } = props.input
-    const [src, srcIsLoading] = usePreviewSrc(
-      value,
-      name,
-      form.getState().values,
-      field.previewSrc
-    )
-
-    let onClear: any
-    if (props.field.clearable) {
-      onClear = () => props.input.onChange('')
-    }
+export const ImageField = wrapFieldsWithMeta<InputProps, ImageProps>(props => {
+  const cms = useCMS()
+  const { form, field } = props
+  const { name, value } = props.input
+  const [src, srcIsLoading] = usePreviewSrc(
+    value,
+    name,
+    form.getState().values,
+    field.previewSrc
+  )
+
+  let onClear: any
+  if (props.field.clearable) {
+    onClear = () => props.input.onChange('')
+  }
 
-    function onChange(media?: Media) {
-      if (media) {
-        props.input.onChange('')
-        props.input.onChange(media)
-      }
+  function onChange(media?: Media) {
+    if (media) {
+      const mediaParse = cms.media.store
+
+      const parsedValue =
+        // @ts-ignore
+        typeof cms?.media?.store?.parse === 'function'
+          ? // @ts-ignore
+            cms.media.store.parse(media)
+          : // @ts-ignore
+            mediaParse.parse(media)
+      props.input.onChange('')
+      props.input.onChange(parsedValue)
     }
+  }
 
-    const uploadDir = props.field.uploadDir || (() => '')
-
-    return (
-      <ImageUpload
-        value={value}
-        previewSrc={src}
-        loading={srcIsLoading}
-        onClick={() => {
-          const directory = uploadDir(props.form.getState().values)
-          cms.media.open({
-            allowDelete: true,
+  const uploadDir = props.field.uploadDir || (() => '')
+
+  return (
+    <ImageUpload
+      value={value}
+      previewSrc={src}
+      loading={srcIsLoading}
+      onClick={() => {
+        const directory = uploadDir(props.form.getState().values)
+        cms.media.open({
```

Comment:
Definitely see what you mean about `media` vs. `media.store`.  Are the APIs available to `media` more global?  As in, the act of `open` is universal, regardless of the `store`?

Reply:
This wasn't a change I made, just a whitespace diff, I probably mixed up the prettier formatter during migration